shared/js/background/classes/privacy-dashboard-data.js

```diff
@@ -0,0 +1,114 @@
+/**
+ * @typedef {import('@duckduckgo/privacy-dashboard/schema/__generated__/schema.types').ExtensionGetPrivacyDashboardData} ExtensionGetPrivacyDashboardData
+ * @typedef {import('@duckduckgo/privacy-dashboard/schema/__generated__/schema.types').DetectedRequest} DetectedRequest
+ * @typedef {import('@duckduckgo/privacy-dashboard/schema/__generated__/schema.types').ProtectionsStatus} ProtectionsStatus
+ * @typedef {import('@duckduckgo/privacy-dashboard/schema/__generated__/schema.types').EmailProtectionUserData} EmailProtectionUserData
+ */
+
+/**
+ * Convert internal extension data into a format accepted by the Privacy dashboard
+ * The return type of this function comes from a schema defined in the Privacy Dashboard,
+ *
+ * @param {import("./tab.es6.js")} tab
+ * @param {EmailProtectionUserData | undefined} userData
+ * @returns {ExtensionGetPrivacyDashboardData}
+ */
+export function dashboardDataFromTab (tab, userData) {
+    const protectionsEnabled = !tab.site.allowlisted && !tab.site.isBroken && tab.site.enabledFeatures.includes('contentBlocking')
+
+    // parent entity, if available
+    let parentEntity
+    if (tab.site.parentEntity) {
+        parentEntity = {
+            displayName: tab.site.parentEntity,
+            prevalence: tab.site.parentPrevalence ?? 0
+        }
+    }
+
+    /** @type {ProtectionsStatus} */
+    const protections = {
+        allowlisted: Boolean(tab.site.allowlisted),
+        denylisted: Boolean(tab.site.denylisted),
+        unprotectedTemporary: Boolean(tab.site.isBroken),
+        enabledFeatures: tab.site.enabledFeatures
+    }
+
+    const requests = convertToRequests(tab, protectionsEnabled)
+
+    return {
+        tab: {
+            id: tab.id,
+            url: tab.url || '',
+            protections,
+            upgradedHttps: tab.upgradedHttps,
+            parentEntity,
+            specialDomainName: tab.site.specialDomainName || undefined
+        },
+        requestData: {
+            requests
+        },
+        emailProtectionUserData: userData
+    }
+}
+
+/**
+ * WIP - this is a just a hack to get the types working together, we'll create a better implementation
+ *
+ * @param {import('./tab.es6')} tab
+ * @param {boolean} protectionsEnabled
+ * @returns {DetectedRequest[]}
+ */
+function convertToRequests (tab, protectionsEnabled) {
+    /** @type {DetectedRequest[]} */
+    const detectedRequests = []
+    for (const tracker of Object.values(tab.trackers || {})) {
+        for (const detectedRequest of Object.values(tracker.urls || {})) {
+            // When protections are off, change the 'state' of each tracking request
+            if (!protectionsEnabled && detectedRequest.action !== 'none') {
+                /** @type {DetectedRequest["state"]} */
+                const nextState = { allowed: { reason: 'protectionDisabled' } }
+                if (!nextState) continue // This should never happen, but if it
```

Comment:
This is assigned on the line before; how is it needed?

Reply:
yep, it was a leftover